### PR TITLE
Updating samples and snippets to NSB.Rabbit 4.3.1

### PR DESCRIPTION
**Pull request by WojcikMike**
Please merge when green

### CODE REVIEW

File: Snippets/Rabbit/Rabbit_4/project.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿{
   "dependencies": {
-    "NServiceBus.RabbitMQ": "4.*"
+    "NServiceBus.RabbitMQ": "4.3.1"
```
Comment (WojcikMike) [inline Snippets/Rabbit/Rabbit_4/project.json]: I think I shouldn't have done that. let me rever this change

Comment (WojcikMike) [inline Snippets/Rabbit/Rabbit_4/project.json]: And done

File: samples/rabbitmq/native-integration/Rabbit_4/Receiver/Receiver.csproj [hunk 2]
```diff
@@ -35,6 +35,9 @@
       <HintPath>..\..\..\..\..\packages\NServiceBus.RabbitMQ.4.3.1\lib\net452\NServiceBus.Transports.RabbitMQ.dll</HintPath>
       <Private>True</Private>
     </Reference>
+    <Reference Include="NServiceBus.Transports.RabbitMQ, Version=4.0.0.0, Culture=neutral, PublicKeyToken=9fc386479f8a226c, processorArchitecture=MSIL">
```
Comment (SimonCropp) [inline samples/rabbitmq/native-integration/Rabbit_4/Receiver/Receiver.csproj]: isnt this a duplicate?

File: samples/rabbitmq/simple/Rabbit_4/Sample/Sample.csproj [hunk 3]
```diff
@@ -35,6 +35,9 @@
       <HintPath>..\..\..\..\..\packages\NServiceBus.RabbitMQ.4.3.1\lib\net452\NServiceBus.Transports.RabbitMQ.dll</HintPath>
       <Private>True</Private>
     </Reference>
+    <Reference Include="NServiceBus.Transports.RabbitMQ, Version=4.0.0.0, Culture=neutral, PublicKeyToken=9fc386479f8a226c, processorArchitecture=MSIL">
+      <HintPath>..\..\..\..\..\packages\NServiceBus.RabbitMQ.4.3.1\lib\net452\NServiceBus.Transports.RabbitMQ.dll</HintPath>
```
Comment (SimonCropp) [inline samples/rabbitmq/simple/Rabbit_4/Sample/Sample.csproj]: isnt this a duplicate?

**Pull request merged**